crates/router/src/connector/noon/transformers.rs

```diff
@@ -416,7 +417,7 @@ impl<F, T>
                     mandate_reference,
                     connector_metadata: None,
                     network_txn_id: None,
-                    connector_response_reference_id: None,
+                    connector_response_reference_id: order.reference,
                 }),
```

Comment:
Hey @sudharsangs !

If we are not getting `reference` in response from Noon, we would like to have a fallback to `order.id`.
So essentially if `reference` is present in response consume that else return the `id` from Noon's response in the `connector_response_reference_id` field.

Reply:
I have made the [change](https://github.com/juspay/hyperswitch/pull/2442/files#diff-76e57cd0c0405611852e98317aa707e02651f9abd8570d0e51adb11732912907R415) hope this is fine.

Reply:
Added comments @sudharsangs